services/device-registry-file/src/main/java/org/eclipse/hono/deviceregistry/file/FileBasedRegistrationService.java

```diff
@@ -707,47 +705,57 @@ public String toString() {
      */
     private String generateDeviceId(final String tenantId) {
 
-        final ConcurrentMap<String, FileBasedDeviceDto> devices = getDevicesForTenant(tenantId);
+        final ConcurrentMap<String, DeviceDto> devices = getDevicesForTenant(tenantId);
         String tempDeviceId;
         do {
             tempDeviceId = UUID.randomUUID().toString();
         } while (devices.containsKey(tempDeviceId));
         return tempDeviceId;
     }
 
-    private static final class FileBasedDeviceDto extends BaseDto<Versioned<Device>> {
+    private static final class FileBasedDeviceDto extends DeviceDto {
 
         FileBasedDeviceDto() {
         }
 
+        public static DeviceDto forRead(final String tenantId, final String deviceId, final JsonObject entry) {
```

Comment:
Yes, I was reluctant to change the signature of DeviceDto's _forUpdate_ method taking a _Supplier_ as this leads to changes in JDBC and MongoDb registry and I didn't want to do this because of a the file based registry which is just for "playing around" 😐.

But as you said below, it seems like this is too confusing. I adapted the implementation to use _FileBasedDeviceDto_ everywhere.